### PR TITLE
Adding sorting algorithms

### PR DESCRIPTION
**Pull request by SatinWuker**
Algorithms Nim seems like a very new project with few contributions. I added two new sorting algorithms, Insertion sort, and Bubble sort. I will try to contribute more algorithms to this project, and I desire to see this project grow!

### CODE REVIEW

**Comment by dlesnoff** [comment]
Please have a look at PR #22 that I made but is a draft right now due to the lack of comments.
I appreciate that you made a lot of comments and provided per file tests.
Precise Best case time complexity rather than just best time complexity.

**Comment by SatinWuker** [comment]
> Please have a look at PR #22 that I made but is a draft right now due to the lack of comments. I appreciate that you made a lot of comments and provided per file tests. Precise Best case time complexity rather than just best time complexity.

@dlesnoff Nice to meet you, and I am really sorry that I did not see your pull request because I made my PR on a train so I wanted to submit it as quickly as possible and did not check if there are similar PRs. However, will this still be merged because of your previous pull request? If not, I will restore my fork and start working on other algorithms.

File: sorting/insertion_sort.nim [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    ## You can treat "key" as a temporary variable
+    var key = arr[i]
+
+    ##  Move elements of arr[0..i-1], that are
+    ##  greater than key, to one position ahead
+    ##  of their current position
+    var j = i - 1
+    while j >= 0 and key < arr[j] :
+      arr[j + 1] = arr[j]
+      j -= 1
```
Comment (dlesnoff) [inline sorting/insertion_sort.nim:33]: Another Nim's syntax sugar that I like. You can resolve this conversation if you do not like this notation.
```suggestion
     dec j
```

File: sorting/insertion_sort.nim [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    arr[j + 1] = key
+
+## Test
```
Comment (dlesnoff) [inline sorting/insertion_sort.nim:36]: Same comments as above.

**Review by dlesnoff — CHANGES_REQUESTED** [review]
Thanks for your kind first contribution, but I think that many issues must be addressed before merging them. I plan to merge your implementation when these are addressed.

In general, the comments duplicates what a developer sees from the code.
They should rather give insights on the «why» of the algorithm. Why two for loops?
Which part of the bubble sort corresponds to a bubble? What can be guaranteed after a «bubble» (a pass) on the list (give invariants)?

Most importantly, it seems like you have overlooked the CONTRIBUTING.md document that we have carefully written to give first contributors an idea of the general structure of a program.
The when isMainModule is important for a file to be shipped inside a big project without messing the imports.

You confused `high(arr)` for `len(arr)`.

The tests do not reflect the genericity of the algorithm. Even if the procedure has been written for integers only, there should at least be a zero and negative numbers in them. Some small random tests would be very much appreciated. Your tests are very similar and could be templated too.

**Comment by SatinWuker** [comment]
I guess I should not do any duplicated work, and I will move on from sorting algorithms to searching algorithms as you are almost done with sortings! Thank you for the problems you pointed out, it really helps me, and I will follow the contribution guidelines, write better comments, and take notice of the code I am writing.

**Pull request closed without merging**